src/_P144_Vindriktning.ino

```diff
@@ -0,0 +1,376 @@
+      //serialHelper_serialconfig_webformLoad(event, true);
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      // this case defines the code to be executed when the form is submitted
+      // the plugin settings should be saved to PCONFIG(x)
+      // ping configuration should be read from CONFIG_PIN1 and stored
+      //serialHelper_webformSave(event);
+
+      // after the form has been saved successfuly, set success and break
+      success = true;
+      break;
+    }
+    case PLUGIN_INIT:
+    {
+      // this case defines code to be executed when the plugin is initialised
+      int8_t rxPin = serialHelper_getRxPin(event);
+      int8_t txPin = serialHelper_getTxPin(event);
+      P144_easySerial = new (std::nothrow) ESPeasySerial(serialHelper_getSerialType(event), rxPin, txPin);
+      if (P144_easySerial != nullptr) 
+      {
+        P144_easySerial->begin(9600);
+        success = true;
+      }
+      String log = F("P144 : Init OK  ESP GPIO-pin RX:");
+      log += rxPin;
+      log += F(" TX:");
+      log += txPin;
+      addLogMove(LOG_LEVEL_INFO, log);
+      break;
+    }
+
+    case PLUGIN_READ:
+    {
+      // code to be executed to read data
+      // It is executed according to the delay configured on the device configuration page, only once
+      UserVar[event->BaseVarIndex] = P144_pm25;
+      String log = F("P144 : READ ");
+      log += P144_pm25;
+      addLogMove(LOG_LEVEL_INFO, log);
+
+      // after the plugin has read data successfuly, set success and break
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WRITE:
+    {
+      // this case defines code to be executed when the plugin executes an action (command).
+      // Commands can be accessed via rules or via http.
+      // As an example, http://192.168.1.12//control?cmd=dothis
+      // implies that there exists the comamnd "dothis"
+
+      break;
+    }
+
+    case PLUGIN_EXIT:
+    {
+      // perform cleanup tasks here. For example, free memory
+      if (P144_easySerial != nullptr) {
+        delete(P144_easySerial);
+        P144_easySerial = nullptr;
+      }
+      success = true;
+      break;
+    }
+
+    case PLUGIN_ONCE_A_SECOND:
+    {
+      // code to be executed once a second. Tasks which do not require fast response can be added here
+      success = true;
+      break;
+    }
+
+    case PLUGIN_TEN_PER_SECOND:
+    {
+      // code to be executed 10 times per second. Tasks which require fast response can be added here
+      // be careful on what is added here. Heavy processing will result in slowing the module down!
+      bool new_data = false;
+      while ((P144_easySerial->available() > 0) && !new_data) 
+      {
+        new_data = P144_process_rx(P144_easySerial->read());
+        if (new_data) 
+        {
+          P144_pm25 = float((P144_serialRxBuffer[3] << 8) + P144_serialRxBuffer[4]);
+          String log = F("P144 : Data ");
```

Comment:
Having that much log statements (10 / second) will make the ESP quite busy, both by building the String, and also sending the (serial or web) data out.
It should at lkeast be wrapped in an `if (loglevelActiveFor(LOG_LEVEL_INFO)) {` but could also be promoted to DEBUG log.
When promoting to DEBUG, you need to add `#ifndef BUILD_NO_DEBUG` / `#endif` around that code, as for some build configurations all DEBUG-related code (including the `#define LOG_LEVEL_DEBUG` !) is removed to save some bytes from the .bin file.

Reply:
This will only trigger once per received message. SO I propose to keep it in, but guarded by #ifdef PLUGIN_144_DEBUG

Reply:
But still the `loglevelActiveFor` check should be applied here too, for those that have logging set to `ERROR` or disabled.